src/spikeinterface/preprocessing/common_reference.py

```diff
@@ -191,10 +191,10 @@ def get_traces(self, start_frame, end_frame, channel_indices):
             else:  # then it must be local
                 re_referenced_traces = np.zeros_like(traces[:, channel_indices])
                 channel_indices_array = np.arange(traces.shape[1])[channel_indices]
```

Comment:
even theses 2 lines looks very very bad. no ?

Reply:
this is because `channel_indices` can be a slice and we need an iterator

Reply:
Maybe they look bad, but they're not breaking anything! Is this ultimately due to the difference between `channel_ids` and `channel_indices`?